contrib/win32/win32compat/w32log.c

```diff
@@ -114,17 +118,37 @@ openlog_file()
 		wcscat_s(ssh_cfg_path, _countof(ssh_cfg_path), __wprogdata); /* "%programData%" */
 		wcscat_s(ssh_cfg_path, _countof(ssh_cfg_path), L"\\ssh"); /* "%programData%\\ssh" */
 
+		wchar_t* tmp_identity = NULL;
+		if (strcmp(identity, "sftp-server") == 0) {
+			tmp_identity = utf8_to_utf16(identity);
+			if (!tmp_identity)
+				return;
+		}
+		else {
+			tmp_identity = malloc((wcslen(tail) - 4) * sizeof(wchar_t));
+			if (!tmp_identity)
+				return;
+			if (wcsncpy_s(tmp_identity, wcslen(tail) - 4, tail + 1, wcslen(tail) - 5) != 0)
+				return;
+		}
+
 		if ((wcsncat_s(log_file, PATH_MAX + 12, ssh_cfg_path, wcslen(ssh_cfg_path)) != 0) ||
 		    (wcsncat_s(log_file, PATH_MAX + 12, logs_dir, 6) != 0) ||
-		    (wcsncat_s(log_file, PATH_MAX + 12, tail + 1, wcslen(tail + 1) - 3) != 0 ) ||
-		    (wcsncat_s(log_file, PATH_MAX + 12, L"log", 3) != 0))
+		    (wcsncat_s(log_file, PATH_MAX + 12, tmp_identity, wcslen(tmp_identity)) != 0) ||
+		    (wcsncat_s(log_file, PATH_MAX + 12, L".log", 4) != 0))
+		{
+			free(tmp_identity);
 			return;
+		}
+
+		free(tmp_identity);
 	}
 	
-	errno_t err = _wsopen_s(&logfd, log_file, O_WRONLY | O_CREAT | O_APPEND, SH_DENYNO, S_IREAD | S_IWRITE);
-		
-	if (logfd != -1)
-		SetHandleInformation((HANDLE)_get_osfhandle(logfd), HANDLE_FLAG_INHERIT, 0);
+	errno_t err;
+	if (strcmp(identity, "sftp-server") == 0)
+		err = _wsopen_s(&sftp_server_logfd, log_file, O_WRONLY | O_CREAT | O_APPEND, SH_DENYNO, S_IREAD | S_IWRITE);
```

Comment:
can you take these in a variable and use it at both line 149, 151.
having a variable makes it consistent behavior and readable.